test/converters/converters.go

```diff
@@ -12,10 +12,11 @@ import (
 	"github.com/bitrise-steplib/steps-deploy-to-bitrise-io/test/junit"
 )
 
-// Intf is the required interface a converter need to match
+// Intf is the required interface a converter needs to match
 type Intf interface {
 	XML() (junit.XML, error)
 	Detect([]string) bool
+	Setup(useOldXCResultExtractionMethod bool)
```

Comment:
Do we need this method for the interface? it is only related to the xcresult converter? Can't we use this information when the xcresult converter is initialised?

Reply:
We are creating the converters as global package level constants here:
https://github.com/bitrise-steplib/steps-deploy-to-bitrise-io/blob/7e83efd64d8060a8141cc40484ca99d17ac3684c/test/converters/converters.go#L22-L26

That means we cannot pass in the user level settings as the processing happens later. So we need to either add a function to the interface and call it before the parsing happens. Or we could try to cast the converter to the `xcresult3` type and when it succeeds we can pass the parameter to it. I choose the first one because it looked nicer.

The ideal would be something as what you wrote that we could customize the individual converter at initialization. I looked at it but it seemed too much of a change for the time being.